src/coreComponents/linearAlgebra/interfaces/hypre/HypreUtils.hpp

```diff
@@ -259,7 +262,247 @@ HYPRE_Int RelaxationSolve( HYPRE_Solver solver,
+{
+  injection = 0,                      //!< Injection \f$[0  I]^{T}\f$
+  jacobi = 2,                         //!< Diagonal scaling
+  classicalModifiedInterpolation = 3, //!< Classical modified interpolation
+  approximateInverse = 4              //!< Approximate inverse
+};
+
+/**
+ * @enum MGRLevelRestrictionType
+ * @brief This enum class specifies the strategy for computing the level restriction operator in MGR.
+ *
+ */
+enum class MGRLevelRestrictionType : HYPRE_Int
+{
+  injection = 0,         //!< Injection \f$[0  I]\f$
+  jacobi = 2,            //!< Diagonal scaling
+  approximateInverse = 3 //!< Approximate inverse
+};
+
+/**
+ * @enum MGRLevelCoarseGridMethod
+ * @brief This enum class specifies the strategy for level coarse grid computation in MGR.
+ */
+enum class MGRLevelCoarseGridMethod : HYPRE_Int
+{
+  galerkin = 0,   //!< Galerkin coarse grid computation using RAP
+  nonGalerkin = 1 //!< Non-Galerkin coarse grid computation with dropping strategy
+};
+
+/**
+ * @enum MGRFRelaxationMethod
+ * @brief This enum class specifies the F-relaxation strategy.
+ */
+enum class MGRFRelaxationMethod : HYPRE_Int
+{
+  singleLevel = 0, //!< single-level relaxation
+  multilevel = 1,  //!< multilevel relaxation
+  amgVCycle = 2    //!< multilevel relaxation
+};
+
+/**
+ * @enum MGRFRelaxationType
+ * @brief This enum class specifies the F-relaxation type.
+ */
+enum class MGRFRelaxationType : HYPRE_Int
+{
+  jacobi = 0,                       //!< Jacobi
+  forwardHybridGaussSeidel = 3,     //!< hybrid Gauss-Seidel or SOR, forward solve
+  backwardHybridGaussSeidel = 4,    //!< hybrid Gauss-Seidel or SOR, backward solve
+  hybridSymmetricGaussSeidel = 6,   //!< hybrid symmetric Gauss-Seidel or SSOR
+  l1hybridSymmetricGaussSeidel = 8, //!< \f$\ell_1\f$-scaled hybrid symmetric Gauss-Seidel
+  l1forwardGaussSeidel = 13,        //!< \f$\ell_1\f$ Gauss-Seidel, forward solve
+  l1backwardGaussSeidel = 14,       //!< \f$\ell_1\f$ Gauss-Seidel, backward solve
+  l1jacobi = 18,                    //!< \f$\ell_1\f$-scaled Jacobi
+  option19 = 19,                    //!< ...
+  option98 = 98                     //!< ...
+};
+
+/**
+ * @enum MGRLevelGlobalSmootherType
+ * @brief This enum class specifies the global smoother type.
+ */
+enum class MGRLevelGlobalSmootherType : HYPRE_Int
+{
+  blockJacobi = 0, //!< block Jacobi (default)
+  jacobi = 1,      //!< Jacobi
+  ilu0 = 16        //!< incomplete LU factorization
+         // nameTBD = 2, //!< Gauss-Seidel, sequential (very slow!)
+         // nameTBD = 3, //!< Gauss-Seidel, interior points in parallel, boundary sequential (slow!)
+         // nameTBD = 4, //!< hybrid Gauss-Seidel or SOR, forward solve
+         // nameTBD = 5, //!< hybrid Gauss-Seidel or SOR, backward solve
+         // nameTBD = 6, //!< hybrid chaotic Gauss-Seidel (works only with OpenMP)
+         // nameTBD = 7, //!< hybrid symmetric Gauss-Seidel or SSOR
+         // nameTBD = 8, //!< Euclid (ILU)
```

Comment:
Are these not defined because they are not yet supported by MGR?

Reply:
The reason for not defining all these options is that we won't use all of them -- for example definitely not the `very slow` ones. I want to have a closer look and ask @oseikuffuor1 for guidance of what to keep. In the meantime, it's probably better to remove them and add new options only when needed.